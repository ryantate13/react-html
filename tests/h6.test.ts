import {h6} from '../src';

describe('h6', () => {
    it('renders with children', () => {
        expect(h6({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(h6({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(h6()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(h6({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(h6({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(h6({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(h6({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(h6({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(h6({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(h6({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(h6({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(h6({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(h6({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(h6({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(h6({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(h6({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(h6({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(h6({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(h6({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(h6({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(h6({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(h6({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(h6({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(h6({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(h6({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(h6({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(h6({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(h6({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(h6({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(h6({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(h6({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(h6({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(h6({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(h6({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(h6({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(h6({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(h6({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(h6({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(h6({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(h6({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(h6({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(h6({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(h6({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(h6({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(h6({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(h6({translate: "no"})).toMatchSnapshot();
    });
});