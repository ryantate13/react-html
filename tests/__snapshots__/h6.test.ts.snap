// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h6 renders with accessKey="a,b,c,d" 1`] = `
<h6
  accessKey="a,b,c,d"
/>
`;

exports[`h6 renders with autoCapitalize="characters" 1`] = `
<h6
  autoCapitalize="characters"
/>
`;

exports[`h6 renders with autoCapitalize="none" 1`] = `
<h6
  autoCapitalize="none"
/>
`;

exports[`h6 renders with autoCapitalize="off" 1`] = `
<h6
  autoCapitalize="off"
/>
`;

exports[`h6 renders with autoCapitalize="on" 1`] = `
<h6
  autoCapitalize="on"
/>
`;

exports[`h6 renders with autoCapitalize="sentences" 1`] = `
<h6
  autoCapitalize="sentences"
/>
`;

exports[`h6 renders with autoCapitalize="words" 1`] = `
<h6
  autoCapitalize="words"
/>
`;

exports[`h6 renders with children 1`] = `
<h6>
  TEST
</h6>
`;

exports[`h6 renders with className="foo bar baz" 1`] = `
<h6
  className="foo bar baz"
/>
`;

exports[`h6 renders with contentEditable="false" 1`] = `
<h6
  contentEditable="false"
/>
`;

exports[`h6 renders with contentEditable="true" 1`] = `
<h6
  contentEditable="true"
/>
`;

exports[`h6 renders with dir="auto" 1`] = `
<h6
  dir="auto"
/>
`;

exports[`h6 renders with dir="ltr" 1`] = `
<h6
  dir="ltr"
/>
`;

exports[`h6 renders with dir="rtl" 1`] = `
<h6
  dir="rtl"
/>
`;

exports[`h6 renders with draggable="false" 1`] = `
<h6
  draggable="false"
/>
`;

exports[`h6 renders with draggable="true" 1`] = `
<h6
  draggable="true"
/>
`;

exports[`h6 renders with empty attributes 1`] = `<h6 />`;

exports[`h6 renders with hidden="false" 1`] = `
<h6
  hidden={false}
/>
`;

exports[`h6 renders with hidden="true" 1`] = `
<h6
  hidden={true}
/>
`;

exports[`h6 renders with id="TEST" 1`] = `
<h6
  id="TEST"
/>
`;

exports[`h6 renders with inputMode="decimal" 1`] = `
<h6
  inputMode="decimal"
/>
`;

exports[`h6 renders with inputMode="email" 1`] = `
<h6
  inputMode="email"
/>
`;

exports[`h6 renders with inputMode="none" 1`] = `
<h6
  inputMode="none"
/>
`;

exports[`h6 renders with inputMode="numeric" 1`] = `
<h6
  inputMode="numeric"
/>
`;

exports[`h6 renders with inputMode="search" 1`] = `
<h6
  inputMode="search"
/>
`;

exports[`h6 renders with inputMode="tel" 1`] = `
<h6
  inputMode="tel"
/>
`;

exports[`h6 renders with inputMode="text" 1`] = `
<h6
  inputMode="text"
/>
`;

exports[`h6 renders with inputMode="url" 1`] = `
<h6
  inputMode="url"
/>
`;

exports[`h6 renders with is="foo" 1`] = `
<h6
  is="foo"
/>
`;

exports[`h6 renders with itemID="https://google.com/" 1`] = `
<h6
  itemID="https://google.com/"
/>
`;

exports[`h6 renders with itemProp="https://google.com/" 1`] = `
<h6
  itemProp="https://google.com/"
/>
`;

exports[`h6 renders with itemRef="foo bar" 1`] = `
<h6
  itemRef="foo bar"
/>
`;

exports[`h6 renders with itemScope="false" 1`] = `
<h6
  itemScope={false}
/>
`;

exports[`h6 renders with itemScope="true" 1`] = `
<h6
  itemScope={true}
/>
`;

exports[`h6 renders with itemType="https://google.com/" 1`] = `
<h6
  itemType="https://google.com/"
/>
`;

exports[`h6 renders with lang="" 1`] = `
<h6
  lang=""
/>
`;

exports[`h6 renders with lang="es-419" 1`] = `
<h6
  lang="es-419"
/>
`;

exports[`h6 renders with slot="TEST" 1`] = `
<h6
  slot="TEST"
/>
`;

exports[`h6 renders with spellCheck="false" 1`] = `
<h6
  spellCheck="false"
/>
`;

exports[`h6 renders with spellCheck="true" 1`] = `
<h6
  spellCheck="true"
/>
`;

exports[`h6 renders with style="[object Object]" 1`] = `
<h6
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`h6 renders with tabIndex="1" 1`] = `
<h6
  tabIndex={1}
/>
`;

exports[`h6 renders with title="TEST" 1`] = `
<h6
  title="TEST"
/>
`;

exports[`h6 renders with translate="no" 1`] = `
<h6
  translate="no"
/>
`;

exports[`h6 renders with translate="yes" 1`] = `
<h6
  translate="yes"
/>
`;

exports[`h6 renders without attributes 1`] = `<h6 />`;
